EFCore.CheckConstraints/Internal/DiscriminatorCheckConstraintConvention.cs

```diff
@@ -22,25 +28,28 @@ public virtual void ProcessModelFinalizing(IConventionModelBuilder modelBuilder,
             foreach (var (rootEntityType, discriminatorValues) in modelBuilder.Metadata
                 .GetEntityTypes()
                 .GroupBy(e => e.GetRootType())
-                .Where(g => g.Key.GetDiscriminatorProperty() != null && g.Key.GetIsDiscriminatorMappingComplete())
+                .Where(g => g.Key.FindDiscriminatorProperty() != null && g.Key.GetIsDiscriminatorMappingComplete())
                 .Select(g => (g.Key, g.Where(e => !e.IsAbstract()).Select(e => e.GetDiscriminatorValue()))))
             {
-                if (!(StoreObjectIdentifier.Create(rootEntityType, StoreObjectType.Table) is StoreObjectIdentifier tableIdentifier))
+                if (!(StoreObjectIdentifier.Create(rootEntityType, StoreObjectType.Table) is { } tableIdentifier))
                 {
                     continue;
                 }
 
-                var discriminatorProperty = rootEntityType.GetDiscriminatorProperty();
+                var discriminatorProperty = rootEntityType.FindDiscriminatorProperty()!;
+
+                // TODO
+                var typeMapping = _typeMappingSource.FindMapping((IProperty)discriminatorProperty);
```

Comment:
/cc @AndriySvyryd, based on an offline conversation we had a long time ago.

This isn't ideal - it won't pick up any explicit Fluent API or DataAnnotation configuration... It may be worth thinking about this. Also note the cast to IProperty, since we only have an IConventionProperty here.

I think this is reasonable for merging as-is, but let me know if you have any comments.

Reply:
I don't remember the conversation, but why not do this?
```suggestion
                var typeMapping = discriminatorProperty.FindTypeMapping() ?? _typeMappingSource.FindMapping((IProperty)discriminatorProperty);
```

Reply:
I can do this, but from playing around, FindTypeMapping was always null in 6.0, while in 5.0 it was always non-null; I think the type mapping only gets populated later now? If I'm missing something, let me know.

Reply:
I'm doing this change in order to release 6.0.0, but it may be good to think about this more - there may be other conventions which need access to property type mappings etc.

Reply:
In 6.0 default type mappings are set after the model has been finalized.

`FindTypeMapping` would return non-null if a type mapping was configured explicitly.

After https://github.com/dotnet/efcore/issues/9329 is fixed `FindTypeMapping` will always return non-null (for supported types)

Reply:
Perfect, thanks @AndriySvyryd